stacks_queues/queue.py

```diff
@@ -10,7 +11,7 @@ class QueueEmptyException(Exception):
 class Queue:
 
     def __init__(self):
-        self.store = [None] * INITIAL_QUEUE_SIZE
+        self.store = LinkedList()* INITIAL_QUEUE_SIZE
```

Comment:
👀 `*` is an unsupported operand for a `LinkedList` and an integer (`INITIAL_QUEUE_SIZE`) meaning that Python doesn't know how to multiply a `LinkedList` object by a number. 

If you're going to implement a queue as a circular buffer, it probably means you'll be using a regular list instead of a linked list. If you want to use a linked list anyways, you wouldn't need to give it an initial buffer size, you can just alter the `LinkedList` class to maintain a tail pointer and point the tail back at the head of the list.

Motivation for why we'd use an array instead of linked list when implementing a circular buffer queue:

It is possible to implement a queue as a circular buffer with a linked list, however if you were to implement a queue with a linked list, you likely wouldn't implement the queue as a circular buffer, since you can insert and remove elements from the head and tail of the list quite easily as long as the linked list is designed to maintain a tail pointer in addition to the head. The amount of memory the linked list takes up will change as you enqueue and dequeue items.


When a queue is implemented with an array, a circular buffer is useful because with an array, when you remove things from the queue either you have to shift all of the remaining elements in the array so that the front of the queue starts at index 0, or you move a head pointer to point at the next index in the list and everything before the new head index is essentially wasted space at the front of the array. 

If you keep adding elements on to the end of the queue implemented with an array, you'll either be taking up a huge amount of space or it will be come very expensive in terms of time to shift the entire array so that the head of the queue is at index 0.